autopush/web/base.py

```diff
@@ -56,22 +59,32 @@ def _validate_request(self, request_handler):
         schema.context["log"] = self.log
         return schema.load(data)
 
-    def _call_func(self, result, func, request_handler, *args, **kwargs):
+    def _call_func(self, result, func, request_handler, start_time, *args,
+                   **kwargs):
         output, errors = result
+        request_handler._timings["validation_time"] = time.time() - start_time
         if errors:
             request_handler._write_validation_err(errors)
         else:
             request_handler.valid_input = output
             return func(request_handler, *args, **kwargs)
 
+    def _track_error_timing(self, failure, request_handler, start_time):
```

Comment:
this could handle both cases of setting the timings as an addBoth "failure" -> "result", and just return result at the end